src/main/java/com/dabsquared/gitlabjenkins/workflow/AcceptGitLabMergeRequestStep.java

```diff
@@ -122,14 +122,13 @@ protected Void run() throws Exception {
         }
 
         private String getCommitMessage(MergeRequest mergeRequest) {
-            if (!step.useMRDescription) return step.mergeCommitMessage;
-
-            String message = "Merge branch '" + mergeRequest.getSourceBranch() + "' into '"
-                    + mergeRequest.getTargetBranch() + "'\n\n" + mergeRequest.getTitle()
-                    + "\n\n" + mergeRequest.getDescription()
-                    + "\n\n" + "See merge request !"
-                    + mergeRequest.getIid();
-            return message;
+            if (!step.useMRDescription || StringUtils.isEmpty(step.mergeCommitMessage)) {
+                return step.mergeCommitMessage;
+            }
+
+            return String.format("Merge branch '%s' into '%s'\n\n%s\n\n%s\n\nSee merge request !%d",
```

Comment:
As far as I can tell from reading the Javadoc, for [format strings](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Formatter.html#syntax), the newline characters need to be replaced with `%n`
```suggestion
            return String.format("Merge branch '%s' into '%s'%n%n%s%n%n%s%n%nSee merge request !%d",
```

Reply:
Yeah, you're indeed correct. `%n` is good choice here. Also it is platform independent and this would be the correct approach for Java's String.format() method. 

Thanks for the clarification. I'll make the changes.